packages/fiori/src/MediaGalleryItem.js

```diff
@@ -0,0 +1,325 @@
+		 * @private
+		 */
+		 focused: {
+			type: Boolean,
+		},
+
+		/**
+		 * @private
+		 */
+		_tabIndex: {
+			type: String,
+			defaultValue: "-1",
+			/* noAttribute: true, */
+		},
+
+		/**
+		 * @private
+		 */
+		 contentHeight: {
+			type: String,
+			noAttribute: true,
+			defaultValue: "",
+		},
+	},
+	slots: /** @lends  sap.ui.webcomponents.fiori.MediaGalleryItem.prototype */ {
+		/**
+		 * Defines the content of the component.
+		 *
+		 * @type {HTMLElement[]}
+		 * @slot
+		 * @public
+		 */
+		 "default": {
+			propertyName: "content",
+			type: HTMLElement,
+		},
+
+		/**
+		 * Defines the content of the thumbnail.
+		 *
+		 * @type {HTMLElement[]}
+		 * @slot
+		 * @public
+		 */
+		 "thumbnail": {
+			propertyName: "thumbnail",
+			type: HTMLElement,
+		},
+	},
+};
+
+/**
+ * @class
+ * <h3 class="comment-api-title">Overview</h3>
+ * The <code>ui5-media-gallery-item</code> web component represents the items displayed in the
+ * <code>ui5-media-gallery</code> web component.
+ * <br><br>
+ * <b>Note:</b> <code>ui5-media-gallery-item</code> is not supported when used outside of <code>ui5-media-gallery</code>.
+ * <br><br>
+ *
+ * <h3>Keyboard Handling</h3>
+ * The <code>ui5-media-gallery</code> provides advanced keyboard handling.
+ * When focused, the user can use the following keyboard
+ * shortcuts in order to perform a navigation:
+ * <br>
+ * <ul>
+ * <li>[SPACE/ENTER/RETURN] - Trigger <code>ui5-click</code> event</li>
+ * </ul>
+ *
+ * <h3>ES6 Module Import</h3>
+ * <code>import "@ui5/webcomponents-fiori/dist/MediaGalleryItem.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.MediaGalleryItem
+ * @extends sap.ui.webcomponents.base.UI5Element
+ * @tagname ui5-media-gallery-item
+ * @public
+ * @implements sap.ui.webcomponents.fiori.IMediaGalleryItem
+ * @since 1.1.0
+ */
+class MediaGalleryItem extends UI5Element {
+	constructor() {
+		super();
+
+		this._thumbnailDesign = !isPhone();
+		this._square = true;
+		this._monitoredContent = null;
+		this._monitoredThumbnail = null;
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return MediaGalleryItemCss;
+	}
+
+	static get template() {
+		return MediaGalleryItemTemplate;
+	}
+
+	get _thumbnail() {
+		return this.thumbnail.length && this.thumbnail[0];
+	}
+
+	get _content() {
+		return this.content.length && this.content[0];
+	}
+
+	get tabIndex() {
+		return this.disabled ? undefined : this._tabIndex;
+	}
+
+	get _useThumbnail() {
+		return this._thumbnailDesign && this.thumbnail.length;
+	}
+
+	get _useContent() {
+		return !this._useThumbnail && this.content.length;
+	}
+
+	get _hideBackgroundIcon() {
+		if (this._useThumbnail) {
+			return !this._thumbnailNotFound;
+		}
+		return !this._contentImageNotFound;
+	}
+
+	onBeforeRendering() {
+		this._monitorLoadingError();
+	}
+
+	onAfterRendering() {
+		this._itemWrapper.style.height = this.contentHeight;
```

Comment:
can you move this to new getter called styles which returns object with keys and then in the template you will call for example styles.itemWrapper